File: LibraryPortal/src/app/model/searchBook.ts
export class SearchBook {
    constructor(private title: string,
                private author: string, private category: string,
                private publishYear: string, private language: string, private role: string) {

        this.title = title;
        this.author = author;
        this.publishYear = publishYear;
        this.category = category;
        this.language = language;
        this.role = role;
    }
}

File: LibraryWebAPI/Models/SearchBook.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;

namespace LibraryApi.Models
{
    public class SearchBook
    {
        public string Title { get; set; }

        public string Author { get; set; }

        public string PublishYear { get; set; }

        public string Category { get; set; }

        public string Language { get; set; }

        public string Role { get; set; }

        public SearchBook()
        {
        }

        public SearchBook(string title,
            string author, string publishYear,
            string category, string language, string role)
        {
            this.Title = title;
            this.Author = author;
            this.PublishYear = publishYear;
            this.Category = category;
            this.Language = language;
            this.Role = role;
        }
    }
}